### DISCUSSION TITLE
Undocumented breaking change introduced in 5.26: removal of /-/all endpoint

### DISCUSSION THREAD

**Discussion by mbargiel** [verdaccio/verdaccio#4224] [Q&A] [answered]
Hello devs!

First of all, I want to say that I'm truly happy to see Verdaccio taking off like this and getting active support, a professional-looking web site, etc. This is great! 😄 

I'd like to report a problem with an undocumented breaking change I discovered today. I found that in Verdaccio 5.26.0, the `/-/all` endpoint was removed. I was able to trace the change back to this commit: https://github.com/verdaccio/verdaccio/commit/43de79d46a9a41313489f30992dcd7cbecddc8ad#diff-725260c5be049ab2bc174657908cd3e53c08876276fb2ded1e0c2f0661a43475

While removing the `/-/all` legacy endpoint is not a bad thing by any means, that removal is still a breaking change, and probably should have been put off to be included in v6.0.0 to avoid a SemVer violation. At the _very least_, I think it should have been documented in an easy-to-find location. (I couldn't find any reference to that removal, but maybe I just didn't look in the right places...) 

In 5.25.0, Verdaccio would log `'/-/all search endpoint is deprecated, might be removed in the next major release'` when it was hit. To my knowledge, 5.26.0 is not a major release 😁 I think the endpoint should have remained until v6.0.0. My suggestion would be to revert the removal of the `/-/all` endpoint in v5.X so that it's only removed in v6.X with a proper note in the changelog.

There's a problem with the replacement, the `/-/v1/search` endpoint, which [does not seem to be fully implemented](https://github.com/verdaccio/verdaccio/issues/3545). That is a major blocker for clients that need to search on registries that have more than 20 packages that can be returned in the same query. For those cases, the `/-/all` could have been an acceptable workaround, but it's no longer available...

**Comment by juanpicado** [comment, marked as answer]
hello, I'm glad to finally meet the unique `/-/all` user that actually uses this endpoint 🙃 and I'm aware of, well I removed by mistake but I'll add it again https://github.com/verdaccio/verdaccio/pull/4233 I was cleaning up :) but seems I did clean up too much, just trying to reuse code between different versions for easy maintenance.

Thanks for letting me know 💯 

> There's a problem with the replacement, the /-/v1/search endpoint, which https://github.com/verdaccio/verdaccio/issues/3545.

yeah that's the reason why I haven't released 6.x yet https://github.com/verdaccio/verdaccio/discussions/4018#discussioncomment-7573430
but :) every time I want to tackle it I never have enough focus time for it.

**Reply by mbargiel** [reply]
Thanks a lot! (And that was quick too! :) )
